fiftyone/utils/sam.py

```
@@ -647,270 +662,317 @@ class SegmentAnythingModel(fout.TorchSamplesMixin, fout.TorchImageModel):
+            samples (None): samples is no longer used. Available for backward compatibility.
 
-        return prompt_field
+        Returns:
+            a list of :class:`fiftyone.core.labels.Detections` instances or a list of dict
+            containing the "masks", "iou_predictions", "low_res_logits" from SAM model output.
+        """
+        if samples is not None:
+            raise RuntimeError(
+                "Use of SamplesMixin has been deprecated. "
+                "Use SegmentAnythingGetItem to get inputs for the predict_all method."
+            )
```

Ruff (0.15.7)

[warning] Avoid specifying long messages outside the exception class

(TRY003)